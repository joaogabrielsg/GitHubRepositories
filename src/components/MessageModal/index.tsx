import React from 'react';
import {StyleSheet, View, Modal, TouchableWithoutFeedback} from 'react-native';

import Text from '../Text';

import {theme, font} from '../../styles';

interface MessageModalProps {
  title: string;
  message: string;
  onClose: () => void;
}

const MessageModal: React.FunctionComponent<MessageModalProps> = props => {
  return (
    <View style={styles.container}>
      <Modal
        animationType="fade"
        transparent
        visible={props.message !== ''}
        onRequestClose={props.onClose}>
        <View style={styles.fullScreenView}>
          <View style={styles.modal}>
            <View style={styles.titleContainer}>
              <Text style={styles.title}>{props.title}</Text>
            </View>
            <View style={styles.messageContainer}>
              <Text style={styles.message}>{props.message}</Text>
            </View>
            <TouchableWithoutFeedback testID="button" onPress={props.onClose}>
              <View style={styles.buttonContainer}>
                <Text style={styles.button}>ok</Text>
              </View>
            </TouchableWithoutFeedback>
          </View>
        </View>
      </Modal>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
    justifyContent: 'center',
    alignItems: 'center',
  },
  fullScreenView: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.modalBackground,
  },
  modal: {
    paddingVertical: 20,
    width: 250,
    flexDirection: 'column',
    justifyContent: 'center',
    alignContent: 'flex-start',
    backgroundColor: 'white',
    borderRadius: 10,
  },
  title: {
    fontSize: font.title,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  titleContainer: {
    paddingVertical: 15,
    width: '100%',
    justifyContent: 'center',
    alignContent: 'center',
  },
  message: {
    fontSize: font.subTitle,
    textAlign: 'center',
  },
  messageContainer: {
    paddingVertical: 15,
    width: '100%',
    justifyContent: 'center',
    alignContent: 'center',
  },
  button: {
    fontSize: font.subTitle,
    textAlign: 'center',
    color: theme.special,
  },
  buttonContainer: {
    paddingVertical: 15,
    width: '100%',
    justifyContent: 'center',
    alignContent: 'center',
  },
});

export default MessageModal;
