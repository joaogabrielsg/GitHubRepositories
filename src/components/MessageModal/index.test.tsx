import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';

import MessageModal from './index';

describe('MessageModal component', () => {
  const onCloseMock = jest.fn();

  it('to match snapshot', () => {
    const {toJSON} = render(
      <MessageModal title="Ops!" message="error test" onClose={onCloseMock} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('to onClose be called on press button', async () => {
    const {getByTestId} = render(
      <MessageModal title="" message="" onClose={onCloseMock} />,
    );

    const button = getByTestId('button');
    fireEvent.press(button);

    expect(onCloseMock).toHaveBeenCalled();
  });
});
