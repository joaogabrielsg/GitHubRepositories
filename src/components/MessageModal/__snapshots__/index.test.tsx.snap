// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageModal component to match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "position": "absolute",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[MockFunction]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(52, 52, 52, 0.7)",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignContent": "flex-start",
            "backgroundColor": "white",
            "borderRadius": 10,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingVertical": 20,
            "width": 250,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "justifyContent": "center",
              "paddingVertical": 15,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Ops!
          </Text>
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "justifyContent": "center",
              "paddingVertical": 15,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            error test
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignContent": "center",
              "justifyContent": "center",
              "paddingVertical": 15,
              "width": "100%",
            }
          }
          testID="button"
        >
          <Text
            style={
              Object {
                "color": "rgb(74, 64, 223)",
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            ok
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
